src/mon/MonClient.cc

```diff
@@ -572,12 +572,11 @@ int MonClient::authenticate(double timeout)
   if (!_opened())
     _reopen_session();
 
-  auto until = ceph::mono_clock::now();
-  until += ceph::make_timespan(timeout);
-  if (timeout > 0.0)
-    ldout(cct, 10) << "authenticate will time out at " << until << dendl;
   while (!active_con && authenticate_err >= 0) {
     if (timeout > 0.0) {
+      auto until = ceph::mono_clock::now();
+      until += ceph::make_timespan(timeout);
+      ldout(cct, 10) << "authenticate will time out at " << until << dendl;
```

Comment:
The entire authentication should time out after five minutes (`timeout` here is `client_mount_timeout` which defaults to 300 seconds).  Have you waited longer than five minutes?  I'm not saying that users would sit there waiting that long, just curious if you saw the timeout fire.

Unless I'm misreading, by moving it into the loop you are actually making it "worse", because the timer is now on the individual wait as opposed to the entire authentication.

Reply:
> The entire authentication should time out after five minutes (`timeout` here is `client_mount_timeout` which defaults to 300 seconds). Have you waited longer than five minutes? I'm not saying that users would sit there waiting that long, just curious if you saw the timeout fire.
> 
indeed, but from what I observed it held lock for more than 5 mins. That's were in successful case we default back to without cephx based authentication and get stuck in this case,
> Unless I'm misreading, by moving it into the loop you are actually making it "worse", because the timer is now on the individual wait as opposed to the entire authentication.

it does makes sense, didn't take it to consideration. But this does not address the problem, which at first I thought it did, it turns out to be sporadic in nature